nix/offchain.nix

```diff
@@ -1,12 +1,463 @@
+
+                  shell = {
+                    withRuntime = true;
+                    packageLockOnly = true;
+                    packages = commandLineTools;
+                  };
+                };
+              in
+              pkgSet;
+
+            bundles = (lib.mapAttrs
+              (name: bundle: project.bundlePursProject {
+                inherit (bundle)
+                  bundledModuleName
+                  webpackConfig;
+                inherit name;
+
+                main = bundle.mainModule;
+                entrypoint = bundle.entrypointJs;
+              })
+              projectConfig.bundles);
+
+            bundleChecks =
+              lib.mapAttrs'
+                (bundleName: _: {
+                  name = "${bundleName}_build-check";
+                  value = bundles.${bundleName};
+                })
+                (lib.filterAttrs
+                  (_: projectBundle: projectBundle.enableCheck)
+                  projectConfig.bundles);
+
+            checks = {
+              bundle-checks =
+                utils.combineChecks "bundle-checks" bundleChecks;
+
+              tests =
+                lib.ifEnable
+                  (projectConfig ? tests)
+                  (project.runPursTest {
+                    inherit (projectConfig.tests)
+                      sources
+                      buildInputs
+                      testMain;
+                  });
+
+              plutip-tests =
+                lib.ifEnable
+                  (projectConfig ? plutip)
+                  (project.runPlutipTest {
+                    inherit (projectConfig.plutip)
+                      sources
+                      buildInputs
+                      testMain;
+                  });
+
+              formatting-check =
+                lib.ifEnable
+                  projectConfig.enableFormatCheck
+                  (pkgs.runCommand "formatting-check"
+                    {
+                      nativeBuildInputs = commandLineTools;
+                    }
+                    ''
+                      cd ${self}
+                      purs-tidy check $(fd -epurs)
+                      nixpkgs-fmt --check $(fd -enix --exclude='spago*')
+                      prettier -c $(fd -ejs)
+                      touch $out
+                    '');
+
+              js-lint-check =
+                lib.ifEnable
+                  projectConfig.enableJsLintCheck
+                  (pkgs.runCommand "js-lint-check"
+                    {
+                      nativeBuildInputs = commandLineTools;
+                    }
+                    ''
+                      cd ${self}
+                      eslint $(fd -ejs)
+                      touch $out
+                    '');
+            };
+
+            ctlRuntimeConfig = projectConfig.runtime.extraConfig // {
+              ctlServer.enable = projectConfig.runtime.enableCtlServer;
+            };
+          in
+          {
```

Comment:
Can we add `nix run` commands? 

- formatting
- pursuit
- updating spago2nix(maybe)

few things I can think of

Reply:
Sure thing, thanks for the suggestion.

Reply:
Note that we already have pursuit, though, with `nix run .#<project-name>_.docs.

Reply:
Done!